### PR TITLE
Update testing

### PR DESCRIPTION
**Pull request by sanderdlm**
I'll open this one as a draft since this ended up being pretty big and includes some drastic changes.

Summary:
[Upgrade the testing workflow](https://github.com/hollodotme/fast-cgi-client/commit/c9842681b575523a9930b15d1208982ca5f60b5c)
 * Dropping PHP 7.1, 7.2, 7.3 and 7.4 from the testing flow
 * Dropping PHPUnit 7 & 8. Moving to a single PHPUnit version (9)
 * Adding PHP 8.2
 * Drop phplint.sh usage
   * PHPStan and PHP-CS (to be added in a later PR?) already offer the same linting benefits. php -ln is redundant in this case.
 * Move phpstan & phpunit from phive installed tooling to a dev dependency.
    * I tried to keep the make/docker set-up as small as possible (just tests) and figured that phpstan is something you want to run locally during development anyway. PHP-CS can also be installed like this.
   
If you don't agree with some changes, feel free to let me know and discuss on the PR:)

### CODE REVIEW

**Comment by sanderdlm** [comment]
Managed to fix the phpstan CI, but tests fail on something unrelated (or I can't figure out what anyway)

**Comment by hollodotme** [comment]
Hey @dreadnip,

first of all: Thank you for your efforts!

I actually don't like to have PHPUnit as a dev dependency, especially if we may need more than one version again in the future. I know that tool management with composer is more convinient, though, and phive does not support multi-version installs either. (At least not officially.)

👍 But for now, I think it's fine to do it that way, as it shrinks down the whole tooling setup.  

👍 Removing the linter is fine.

👍 I'd love to see PHP-CS in the pipeline. So if you find the time to add it, feel free to do so, but it should be a separate PR, please.

🖊️ I think we can also require `ext-xdebug` 3.x now in composer.json.

File: composer.json [hunk 1]
```diff
@@ -27,10 +27,12 @@
         }
     },
     "require": {
-        "php": ">=7.1",
+        "php": ">=8.0",
         "ext-json": "*"
     },
     "require-dev": {
-        "ext-xdebug": ">=2.6.0"
+        "ext-xdebug": ">=2.6.0",
```
Comment (hollodotme) [inline composer.json]: I think we can raise the version constraint to ^3.1 here for ext-xdebug

**Review by hollodotme — CHANGES_REQUESTED** [review]
Did not find anything problematic, great work!

Please change the ext-xdebug constraint and we can merge it.

**Comment by hollodotme** [comment]
> Managed to fix the phpstan CI, but tests fail on something unrelated (or I can't figure out what anyway)

I can have a look at the tests after merging.

File: composer.json (hunk 1 above)
Comment (sanderdlm) [reply composer.json]: Bumped in 715354a5012f5b16a49c6da7cee0bd866121b458

**Comment by sanderdlm** [comment]
Rebased on 4.x-dev with typed properties.

**Review by hollodotme — APPROVED** [review]
Looks good. Thanks! I'll have a look at the tests later today or tomorrow.

**Pull request merged**